go/pkg/feed/service.go

```diff
@@ -171,7 +171,13 @@ func (s *FeedService) Posts(ctx context.Context, req *feedpb.PostsRequest) (*fee
 	posts := make([]*feedpb.Post, len(dbPostWithExtras))
 	for idx, dbPost := range dbPostWithExtras {
 		var reactions []*feedpb.Reaction
-		reactionsMap := map[string]interface{}{}
+
+		reactionsMap := make(map[string]interface{})
+		if err := json.Unmarshal(dbPost.UserReactions, &reactionsMap); err != nil {
+			s.conf.Logger.Error("failed to unmarshal UserReactions", zap.String("data", string(dbPost.UserReactions)), zap.Error(err))
+			continue
+		}
+
 		dbPost.UserReactions.Scan(&reactionsMap)
```

Comment:
I think the problem with scan comes from the change of type of from the custom jsonb encoder to the `datatypes.JSON` gorm type, we should use the gorm type IMO

Reply:
Ok, so is it ok ? https://github.com/TERITORI/teritori-dapp/pull/1396/commits/f1bdc8283137edc4901a378468121edf39f975ef

Reply:
I think json.Unmarshal will just unmarshal the data with defined struct so as long as it's compatible, it will work, that's the reason why the code works with json.Unmarshal (Scan might do some type check also so it does not work) because the root cause might be related to this:
<img width="1352" alt="Screenshot 2024-11-20 at 10 14 15 PM" src="https://github.com/user-attachments/assets/39d61400-d31f-4784-946e-7eb29cfe36ed">

so IMHO, the fix is valid, it can work but for the proper solution you should check the types change.

Reply:
@hthieu1110 I don't know what to do when I see these 2 typpes changes and this TODO:
https://github.com/TERITORI/teritori-dapp/blob/fix-post-reactions-from-feed/go/internal/indexerdb/jsonb.go#L10

Reply:
Btw, look at this TODO: https://github.com/TERITORI/teritori-dapp/blob/fix-post-reactions-from-feed/go/internal/indexerdb/jsonb.go#L10
If we fix the TODO, `UserReactions` gets the type `datatypes.JSON` finally